tracing-core/src/event.rs

```diff
@@ -79,6 +79,24 @@ impl<'a> Event<'a> {
         });
     }
 
+    /// Constructs a new `Event` with the specified metadata and set of values,
+    /// and returns whether such an event would be enabled by the current collector.
+    pub fn child_of_enabled(
+        parent: impl Into<Option<Id>>,
+        metadata: &'static Metadata<'static>,
+        fields: &'a field::ValueSet<'_>,
+    ) -> bool {
+        let event = Self::new_child_of(parent, metadata, fields);
+        crate::dispatch::get_default(|current| current.enabled(event.metadata()))
+    }
+
+    /// Constructs a new `Event` with the specified metadata and set of values,
+    /// and returns whether such an event would be enabled by the current collector.
+    pub fn enabled(metadata: &'static Metadata<'static>, fields: &'a field::ValueSet<'_>) -> bool {
+        let event = Event::new(metadata, fields);
+        crate::dispatch::get_default(|current| current.enabled(event.metadata()))
+    }
```

Comment:
I don't think that the enabled macro should create a new `Event` variant; that seems incorrect to me. `Subscriber::enabled` Instead, I think it should just create a callsite and metadata, and call `Dispatch::enabled` with that metadata. I guess the intention was to ask the subscriber if it would enable an event with specific field _values_...but filtering isn't performed on field values anyway, so there's no point in having this take field values.

Similarly, I don't believe `child_of_enabled` should exist, as it does nothing. `enabled` is only called with `Metadata`, and `Metadata` doesn't include the parent span.

Reply:
should we just not support `parent: ` in the `enabled!` macro?

Reply:
yeah, the macro shouldn't accept a `parent` or field values.